File: arrays/merge_arrays/merge_arrays.py
def merge_arrays(first, second):
  first_idx = 0
  second_idx = 0
  first_arr_length = len(first)
  second_arr_length = len(second)

  merged_array = []
  while (first_idx < first_arr_length and second_idx < second_arr_length):
    first_ele = first[first_idx]
    second_ele = second[second_idx]

    if first_ele <= second_ele:
      merged_array.append(first_ele)
      first_idx += 1
    else:
      merged_array.append(second_ele)
      second_idx += 1
  
  # Cleanup any array that hasnt been merged
  if (first_idx < first_arr_length):
    # first array needs to be completely merged
    while (first_idx < first_arr_length):
      ele = first[first_idx]
      merged_array.append(ele)
      first_idx += 1
  else: 
    while (second_idx < second_arr_length):
      ele = second[second_idx]
      merged_array.append(ele)
      second_idx += 1
  
  return merged_array



my_list     = [3, 4, 6, 10, 11, 15]
alices_list = [1, 5, 8, 12, 14, 19]

# Prints [1, 3, 4, 5, 6, 8, 10, 11, 12, 14, 15, 19]
print(merge_arrays(my_list, alices_list))

File: arrays/merge_arrays/merge_arrays.js
// Two arrays of orders sorted numerically
// Write function to merge our array of orders into one sorted array

// const myArray = [3, 4, 6, 10, 11, 15];
// const alicesArray = [1, 5, 8, 12, 14, 19];

// console.log(mergeArrays(myArray, alicesArray));
// logs [1, 3, 4, 5, 6, 8, 10, 11, 12, 14, 15, 19]

function mergeArrays(firstArray, secondArray) {
  let firstIdx = 0;
  let secondIdx = 0;
  const firstArrLength = firstArray.length;
  const secondArrLength = secondArray.length;

  const mergedArr = [];
  while (firstIdx < firstArrLength && secondIdx < secondArrLength) {
    const firstEle = firstArray[firstIdx];
    const secondEle = secondArray[secondIdx];
    if (firstEle <= secondEle) {
      // smaller or equal so add it into mergedArr first
      mergedArr.push(firstEle);
      firstIdx++;
    } else {
      // add secondEle since it is smaller
      mergedArr.push(secondEle);
      secondIdx++;
    }
  }

  // Final cleanup: check if any of the arrays did not reach the final index
  if (firstIdx < firstArrLength) {
    while (firstIdx < firstArrLength) {
      const ele = firstArray[firstIdx];
      mergedArr.push(ele);
      firstIdx++;
    }
  } else {
    while (secondIdx < secondArrLength) {
      const ele = secondArray[secondIdx];
      mergedArr.push(ele);
      secondIdx++;
    }
  }

  return mergedArr;
}

const myArray = [3, 4, 6, 10, 11, 15];
const alicesArray = [1, 5, 8, 12, 14, 19];

console.log(mergeArrays(myArray, alicesArray));